File: desafios/entreLetras.js
//https://dojopuzzles.com/problems/entre-letras/

const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

const entreLetras = (letra1, letra2) => { 
    letras = ["A" , "B" , "C" , "D" , "E" , "F" , "G" , "H" , "I" , "J" , "K" , "L" , "M" , "N" , "O" , "P" , "Q" , "R" , "S" , "T" , "U" , "V" , "W" , "X" , "Y" , "Z"]

    let indexLetra1 = letras.indexOf(letra1.toUpperCase()) + 1;
    let indexLetra2 = letras.indexOf(letra2.toUpperCase());

    console.log(indexLetra1);
    console.log(indexLetra2);


    if (indexLetra1 >= indexLetra2) return console.log('Não está na ordem alfabética');

    console.log(`${letra1}, ${letra2} =  ${indexLetra2 - indexLetra1}`);

}

rl.question('Insira as duas letras em ordem alfabética separadas por espaço: ', (entrada) => {
    
    let letras = entrada.replace(/\s+/g, '').split('')
    console.log(letras);
    entreLetras(letras[0], letras[1]);
   
    rl.close();
});

File: entre_letras.py
def letras(letra1, letra2):
    alfabeto = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m",
                "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]
    if alfabeto.index(letra1) > alfabeto.index(letra2):
        print("Digite em ordem alfabética.")
    else:
        distancia = alfabeto.index(letra2) - alfabeto.index(letra1) -1
        print("'{letra1}','{letra2}' = {distancia}".
            format(letra1 = letra1, letra2=letra2, distancia=distancia))
